apps/website/src/utils/datetime.ts

```diff
@@ -71,6 +71,48 @@ export const formatDateTimeParts = (
     .reconfigure({ locale: locale ?? undefined })
     .toLocaleParts(getFormat({ style, time, timezone }));
 
+export const formatDateTimeRelative = (
+  dateTime: Date,
+  {
+    style = "short",
+    time = undefined,
+    timezone = false,
+  }: Partial<DateTimeFormat> = {},
+  { locale = "en-US", zone = "UTC" }: Partial<DateTimeOptions> = {},
+) => {
+  // Format the date
+  const parts = formatDateTimeParts(
+    dateTime,
+    { style, time, timezone },
+    { locale, zone },
+  );
+
+  // Determine how many days away the date is
+  const dateToday = DateTime.now().setZone(zone ?? undefined);
+  const dateGiven = DateTime.fromJSDate(dateTime).setZone(zone ?? undefined);
+  const daysToday = dateToday.startOf("day").toUnixInteger() / (60 * 60 * 24);
+  const daysGiven = dateGiven.startOf("day").toUnixInteger() / (60 * 60 * 24);
```

Comment:
I have the feeling unix timestamps of the start of the day might not always be divisible by 86400 which would break the `=== 0`/`=== 1`? But I might be wrong. I hate time ;)

Reply:
The start of the day is always going to be 0 seconds into the day, and the epoch started at 0 seconds into a day, so it should always be divisible by 86400 (number of seconds in a day).

Reply:
So it probably will only happen on the days where daylight savings changes (since leap seconds seem to not a thing anymore).

```typescript
const dateToday = DateTime.fromJSDate(new Date(2024, 2, 11)).setZone("America/Chicago");
const dateGiven = DateTime.fromJSDate(new Date(2024, 2, 12)).setZone("America/Chicago");
const daysToday = dateToday.startOf("day").toUnixInteger() / (60 * 60 * 24);
const daysGiven = dateGiven.startOf("day").toUnixInteger() / (60 * 60 * 24);

console.log({ daysGiven, daysToday, days: daysGiven - daysToday });
  
`{daysGiven: 19793.208333333332, daysToday: 19792.25, days: 0.9583333333321207}`
```

Reply:
Oh ew I forgot about daylight savings -- I think it'd be safe to just floor these once the division is done?

Reply:
Yeah! Maybe a plain round to get both ways (rounding up and down)?

Reply:
Where would rounding up ever be valid? We always want to get the days up to that day, not including that day?

Reply:
Yeah i was thinking about rounding the difference, not the terms. This is great :)